### PR TITLE
add more test cases for default runtime validation

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

### CODE REVIEW

File: validation/validation_test.go [hunk 1]
```diff
@@ -88,13 +88,44 @@ func TestValidateBasic(t *testing.T) {
...
+func TestValidateHostname(t *testing.T) {
+	g := getDefaultGenerator()
+	g.SetHostname("hostname-specific")
+
+	assert.Nil(t, runtimeInsideValidate(g))
+}
+
+// Test whether mounts are correctlt mounted
+func TestValidateMounts(t *testing.T) {
+	// TODO mounts generation options have not been implemented
```
Comment (liangchenye) [inline validation/validation_test.go]: we can add this test when 'mounts generate' will be done.

File: validation/validation_test.go [hunk 2]
```diff
@@ -88,13 +88,44 @@ func TestValidateBasic(t *testing.T) {
 	assert.Nil(t, runtimeInsideValidate(g))
 }
 
+// Test whether rootfs Readonly can be applied or not
+func TestValidateRootFS(t *testing.T) {
+	g := getDefaultGenerator()
+	g.SetRootReadonly(true)
```
Comment (liangchenye) [inline validation/validation_test.go]: Maybe we can add another test like:
```
g.SetRootReadonly(false)
```

Comment (Mashimiao) [reply validation/validation_test.go]: Added, PTAL

File: validation/validation_test.go (hunk 1 above)
Comment (Mashimiao) [reply validation/validation_test.go]: Added such TODO comment

File: validation/validation_test.go [hunk 3]
```diff
@@ -88,13 +88,53 @@ func TestValidateBasic(t *testing.T) {
...
+	g := getDefaultGenerator()
+	g.SetRootReadonly(false)
+
+	assert.Nil(t, runtimeInsideValidate(g))
+}
+
+// Test whether rootfs Readonly can be applied as true
+func TestValidateRootFS(t *testing.T) {
+	g := getDefaultGenerator()
+	g.SetRootReadonly(true)
```
Comment (wking) [inline validation/validation_test.go]: I think we need to skip this on Windows, where [the spec forbids setting `readonly` to `true`][1].  Also, you probably need to use more-specific function names:

```
$ git grep '^func TestValidateRootFS' origin/pr/512
origin/pr/512:validation/validation_test.go:func TestValidateRootFS(t *testing.T) {
origin/pr/512:validation/validation_test.go:func TestValidateRootFS(t *testing.T) {
```

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.0/config.md#L46

File: validation/validation_test.go [hunk 4]
```diff
@@ -88,13 +88,53 @@ func TestValidateBasic(t *testing.T) {
...
+
+// Test whether hostname can be applied or not
+func TestValidateHostname(t *testing.T) {
+	g := getDefaultGenerator()
+	g.SetHostname("hostname-specific")
+
+	assert.Nil(t, runtimeInsideValidate(g))
+}
+
+// Test whether mounts are correctlt mounted
```
Comment (wking) [inline validation/validation_test.go]: “correctlt” → “correctly”.  Or just use `// Validate mounts`, or leave the comments off?  `TestValidateMounts` sounds pretty clear to me even without a comment.

**Comment by Mashimiao** [comment]
updated. @wking  @liangchenye  @q384566678  PTAL

**Comment by liangchenye** [comment]
LGTM

File: cmd/runtimetest/main.go [hunk 5]
```diff
@@ -336,6 +336,11 @@ func validateRootFS(spec *rspec.Spec) error {
 		if err == nil {
 			return specerror.NewError(specerror.RootReadonlyImplement, fmt.Errorf("rootfs must be readonly"), rspec.Version)
 		}
+	} else {
+		err := testWriteAccess("/")
+		if err == nil {
```
Comment (q384566678) [inline cmd/runtimetest/main.go]: Should use `err != nil`.

Comment (Mashimiao) [reply cmd/runtimetest/main.go]: good catch, updated

**Comment by q384566678** [comment]
LGTM

**Comment by liangchenye** [comment]
LGTM

**Pull request merged**